src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1227,13 +1227,20 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
             int arity = node.Arity;
             bool hasTypeArguments = arity > 0;
 
-            SeparatedSyntaxList<TypeSyntax> typeArgumentList = node.Kind() == SyntaxKind.GenericName
-                ? ((GenericNameSyntax)node).TypeArgumentList.Arguments
-                : default(SeparatedSyntaxList<TypeSyntax>);
+            GenericNameSyntax genericName;
+            SeparatedSyntaxList<TypeSyntax> typeArgumentList = default(SeparatedSyntaxList<TypeSyntax>);
+            bool isUnboundTypeInNameof = false;
+
+            if (node.Kind() == SyntaxKind.GenericName)
```

Comment:
Test failure is due to cascade diagnostic on malformed type argument lists. We probably need to return early when `node.ContainsDiagnostic` is true.